vulnerabilities/package_managers.py

```diff
@@ -678,3 +682,41 @@ def fetch(self, pkg: str) -> Iterable[PackageVersion]:
             version = self.fetch_version_info(version_info, escaped_pkg)
             if version:
                 yield version
+
+
+VERSION_API_CLASSES = {
+    MavenVersionAPI,
+    NugetVersionAPI,
+    ComposerVersionAPI,
+    PypiVersionAPI,
+    RubyVersionAPI,
+    GoproxyVersionAPI,
+    NpmVersionAPI,
+    HexVersionAPI,
+    LaunchpadVersionAPI,
+    CratesVersionAPI,
+    DebianVersionAPI,
+    GitHubTagsAPI,
+}
+
+VERSION_API_CLASSES_BY_PACKAGE_TYPE = {cls.package_type: cls for cls in VERSION_API_CLASSES}
+
+
+def get_api_package_name(purl: PackageURL) -> str:
+    """
+    Return the package name expected by the GitHub API given a PackageURL
+    >>> get_api_package_name(PackageURL(type="maven", namespace="org.apache.commons", name="commons-lang3"))
+    'org.apache.commons:commons-lang3'
+    >>> get_api_package_name(PackageURL(type="composer", namespace="foo", name="bar"))
+    'foo/bar'
+    """
+    if purl.type == "maven":
+        return f"{purl.namespace}:{purl.name}"
+
+    if purl.type == "composer":
+        return f"{purl.namespace}/{purl.name}"
+
+    if purl.type in ("nuget", "pypi", "gem", "golang", "npm"):
+        return purl.name
```

Comment:
This will unlikely be correct for goland and npm as both can have namespaces.

Reply:
@pombredanne good catch, adding to your point, some packages may not have a namespace so in that case we can directly return name for example this https://registry.npmjs.org/hapi and those who may have namespace can be clubbed using '/' like this https://registry.npmjs.org/@curveball/a12n-server